src/components/DataEntry/DataEntry.jsx

```diff
@@ -0,0 +1,112 @@
+import React, { Component } from 'react';
+import './DataEntry.css';
+
+class DataEntry extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      tablenum: '',
+      clusternum: '',
+      wavenum: ''
+    };
+    this.handleTable = this.handleTable.bind(this);
+    this.handleCluster = this.handleCluster.bind(this);
+    this.handleWave = this.handleWave.bind(this);
+    this.saveVals = this.saveVals.bind(this);
+  }
+
+  handleTable(event) {
+    this.setState({
+      tablenum: event.target.value
+    });
+    if (event.key === 'Enter') {
+      console.log(this.state.tablenum);
```

Comment:
this needs to be event.target.value

the state isn't updated in time for `this.state.tablenum` to output the correct result. Check you console log to see the difference when you have `event.target.value` and `this.state.tablenum` here